front/app/modules/commercial/insights/hooks/useScanInsightsCategory.tsx

```diff
@@ -0,0 +1,176 @@
+import { useEffect, useState, useRef } from 'react';
+
+// services
+import {
+  insightsCategoriesSuggestionsTasksStream,
+  insightsTriggerCategoriesSuggestionsTasks,
+} from 'modules/commercial/insights/services/insightsCategoriesSuggestionsTasks';
+
+// tracking
+import { trackEventByName } from 'utils/analytics';
+import tracks from 'modules/commercial/insights/admin/containers/Insights/tracks';
+
+import { BehaviorSubject } from 'rxjs';
+import { tap, delay } from 'rxjs/operators';
+
+import streams from 'utils/streams';
+
+export type ScanStatus =
+  | 'isIdle'
+  | 'isInitializingScanning'
+  | 'isScanning'
+  | 'isFinished';
+
+type ScanProps = Record<
+  string,
+  {
+    status: ScanStatus;
+    initialTasksCount: number;
+    completedTasksCount: number;
+  }
+>;
+
+const $scanCategory: BehaviorSubject<ScanProps> = new BehaviorSubject({});
```

Comment:
We are now using a progress indicator to show the scanning progress and refetching the inputs while polling. For this reason, I have opted for putting the polling back in the hook and only storing the scan status in a global observable that I can access from anywhere in the app. The reason is that we most probably will need this status in places where the hook is not necessarily being used.